core/src/display_object/movie_clip.rs

```diff
@@ -2106,13 +2112,18 @@ impl<'gc> TInteractiveObject<'gc> for MovieClip<'gc> {
                             }
                         }
 
-                        context.action_queue.queue_actions(
-                            self.into(),
-                            ActionType::Normal {
-                                bytecode: event_handler.action_data.clone(),
-                            },
-                            event == ClipEvent::Unload,
-                        );
+                        // Some SWF defined-event handlers must be queued after the ActionScript defined ones.
+                        if event == ClipEvent::Data {
```

Comment:
I wonder if it's a little better to copy + paste, say, the AS-defined dispatch section instead of pushing to a queue:
```
if event == ClipEvent::Data {
  // Fire SWF-defined event
}

if let Some(flag) = event.flag() {
  // Fire SWF-defined event
}

if event != ClipEvent::Data {
  // Fire AS-defined event
}
```
But maybe this is a bit messier.